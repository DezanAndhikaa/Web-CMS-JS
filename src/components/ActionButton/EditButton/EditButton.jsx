import React from 'react';
import InputText from '../../InputText/InputText'
import { Modal } from '@material-ui/core'
import './EditButton.scss';

class EditButton extends React.PureComponent {

	constructor(props){
        super(props);
        this.state={
            isShowModal: false
        }
    }

    isClicked = () => {
        this.setState({isShowModal: !this.state.isShowModal})
    }

    isCloseds = () => {
        this.setState({isShowModal: !this.state.isShowModal})
    }

	render() {
        // console.log("visi visi visi ",this.props.values)
		return (
			<div className="edit-button-row">
				<div className={this.props.values  > 0 ? "edit-button-visible" : "edit-button-unvisible"} onClick={this.isClicked} />
					<Modal 
					className="modal-pos" open={this.state.isShowModal} onClose={this.isCloseds}>
                        <div>
                            <InputText 
                                {...this.props}
                                {...this.state}
                                onClosed={this.isCloseds}
                            />
                        </div>
                    </Modal>
			</div>
		);
	}
}

export default EditButton;