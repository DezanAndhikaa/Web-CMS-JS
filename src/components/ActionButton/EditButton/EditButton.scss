@import '../../../themes/functions';
@import '../../../themes/variables';

.edit-button-row{
    display: flex;
    justify-content: flex-end;
    height: vw(30);
    width: 100%;
    .edit-button{
        height: vw(30);
        width: vw(30);
        background-image: url('../../../assets/icons/edit.png');
        background-repeat: no-repeat;
        background-size: contain;
    }
}

@media (min-width: 600px) and (max-width: 1025px) {
    .edit-button-row{
        display: flex;
        justify-content: flex-end;
        height: vw-tab(30);
        width: 100%;
        .edit-button{
            height: vw-tab(30);
            width: vw-tab(30);
            background-image: url('../../../assets/icons/edit.png');
            background-repeat: no-repeat;
            background-size: contain;
        }
    }
}

@media (max-width: 599px) {    
    .edit-button-row{
        display: flex;
        justify-content: flex-end;
        height: vw-mobile(20);
        width: 100%;
        .edit-button{
            height: vw-mobile(20);
            width: vw-mobile(20);
            background-image: url('../../../assets/icons/edit.png');
            background-repeat: no-repeat;
            background-size: contain;
        }
    }
}
