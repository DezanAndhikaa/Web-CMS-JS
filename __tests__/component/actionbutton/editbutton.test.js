import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import EditButton from "../../../src/components/ActionButton/EditButton/EditButton";

Enzyme.configure({adapter:new Adapter()});
it("render edit button correctly", () => {
  const wrapper = shallow (
    <EditButton />
  )
  expect(wrapper).toMatchSnapshot();
})