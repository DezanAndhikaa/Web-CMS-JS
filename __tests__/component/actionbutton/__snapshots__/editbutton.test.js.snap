// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render edit button correctly 1`] = `""`;
